app/reflexes/admin/connected_app_reflex.rb

```diff
@@ -3,10 +3,43 @@
 module Admin
   class ConnectedAppReflex < ApplicationReflex
     def create
-      enterprise = Enterprise.find(element.dataset.enterprise_id)
       authorize! :admin, enterprise
+
       app = ConnectedApp.create!(enterprise_id: enterprise.id)
 
+      # Avoid race condition by sending before enqueuing job:
+      broadcast_partial
+
+      ConnectAppJob.perform_later(
+        app, current_user.spree_api_key,
+        channel: SessionChannel.for_request(request),
+      )
+      morph :nothing
+    end
+
+    def destroy
+      authorize! :admin, enterprise
+
+      app = enterprise.connected_apps.first
```

Comment:
Correct me if I am wrong, currently we can only have one connected app, and it's hardcoded right ? maybe we should have a comment here to clarify that.

Reply:
That's right.